### PR TITLE
P2P networking layer draft

### PR DESCRIPTION
**Pull request by greenhat**
Ref #481

This PR partially drafts p2p networking layer. 
Missing parts (to be implemented later):
- exchange of the handshake messages;
- manage open TCP connections;

### CODE REVIEW

File: ergo-p2p/Cargo.toml [hunk 1]
```diff
@@ -14,14 +14,18 @@ exclude = [
...
 [dependencies]
-
 sigma-ser = { version = "^0.3.0", path = "../sigma-ser" }
 sigma-util = { version = "^0.3.0", path = "../sigma-util" }
 thiserror = "1"
 derive_more = "0.99"
 bounded-vec = { version = "^0.5.0" }
 proptest-derive = {version = "0.3.0", optional = true }
 
+futures = "0.3"
```
Comment (ross-weir) [inline ergo-p2p/Cargo.toml:24]: Super early in the draft but just wanted to mention:

`tokio` might be a good choice for the networking/async layer if you haven't looked at it 😃

Comment (greenhat) [reply ergo-p2p/Cargo.toml:24]: Oh, there will be `tokio` in it for sure. :) Thanks for checking in!

**Comment by greenhat** [comment]
I'm switching to REST-based peer messaging for the SPV client https://github.com/ergoplatform/sigma-rust/issues/495 and I want to merge it since I have no immediate plan to continue working on it.

File: ergo-p2p/src/client.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#![allow(dead_code)] // TODO: remove
+
+use futures::channel::mpsc;
+use futures::channel::oneshot;
```
Comment (ross-weir) [inline ergo-p2p/src/client.rs:4]: Out of curiosity, any particular reason we're going with `futures-rs` for these over `tokio`?

Had a bit of a google for comparisons but didn't find that much apart from `futures-rs` being developed by the `rust` team

**Review by ross-weir — APPROVED** [review]

File: ergo-p2p/src/client.rs (hunk 2 above)
Comment (greenhat) [reply ergo-p2p/src/client.rs]: Good catch! No reason apart from it was used in Zebra. Switched to tokio's.

**Pull request merged**